src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -369,6 +403,13 @@ public partial class ProjectResourceOptions
         public string? LaunchProfileName { get { throw null; } set { } }
     }
 
+    public static partial class PublisherDistributedApplicationBuilderExtensions
+    {
+        [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPUBLISHERS001")]
+        public static void AddPublisher<TPublisher, TPublisherOptions>(this IDistributedApplicationBuilder builder, string name, System.Action<TPublisherOptions>? configureOptions = null)
```

Comment:
`AddPublisher` should return `IDistributedApplicationBuilder`. It's experimental but if we're still taking changes...

Reply:
https://github.com/dotnet/aspire/pull/8665